### PR TITLE
[mesheryctl] Gracefully handle errors when platform is unavailable

### LINKED ISSUE #3558: [mesheryctl] `system status` errors when platform dependency is missing
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior
mesheryctl presents an error when executing `system status` when the system is missing prerequisite dependencies. The system being used in this scenario does not have Docker installed (does not have `docker` available.

#### Desired Behavior
mesheryctl should detect whether or not prerequisites of the platform are availablle (`docker` cli in this case) and capture this error gracefully, informing the user of the failed preflight check (of the specific missing dependency).

Preflight health check should identify the missing dependency.

#### Screenshots/Logs
```
➜  mesheryctl git:(master) ./mesheryctl system status
Missing Meshery config file.
Create default config now [y/n]? y
Default config file created at /Users/willcalcote/.meshery/config.yaml
Error: exec: "docker-compose": executable file not found in $PATH
Usage:
  mesheryctl system status [flags]

Flags:
  -h, --help   help for status

Global Flags:
      --config string    config file (default location is: /Users/willcalcote/.meshery/config.yaml) (default "/Users/willcalcote/.meshery/config.yaml")
  -c, --context string   (optional) temporarily change the current context.
  -v, --verbose          verbose output
  -y, --yes              (optional) assume yes for user interactive prompts.
```

#### Environment

- Host OS: Mac M1
- Meshery Client Version: stable-v0.5.37
- Platform: neither

#### Contributor Resources

- [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by alphaX86**
Signed-off-by: Aadhitya A <aadhitya864@gmail.com>

**Description**

This PR fixes #3558 

**Notes for Reviewers**
This PR is still WIP

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/pkg/utils/healthcheck.go [hunk 1]
```diff
@@ -119,7 +119,7 @@ func IsMesheryRunning(currPlatform string) (bool, error) {
 		{
 			op, err := exec.Command("docker-compose", "-f", DockerComposeFile, "ps").Output()
 			if err != nil {
-				return false, err
+				return false, errors.Wrap(err, " docker-compose isn't available or down for a moment. Please run mesheryctl system check --preflight for system check")
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/healthcheck.go:122]:
```suggestion
				return false, errors.Wrap(err, " required dependency, docker-compose, is not present or docker is not available. Please run `mesheryctl system check --preflight` to verify system readiness")
```

**Comment by alphaX86** [comment]
@leecalcote just a doubt... is the go version fixed? I'm running issues on CI coz of that

**Comment by alphaX86** [comment]
Hmm... let's see if I go one step back....

**Comment by navendu-pottekkat** [comment]
@alphaX86 The lint issue is real. The rest fails due to the Go version error. It has been fixed but for some reason the workflow is not updated.

**Comment by alphaX86** [comment]
> @alphaX86 The lint issue is real. The rest fails due to the Go version error. It has been fixed but for some reason the workflow is not updated.

Oh ok... for some unknown reason I thought on why it's failing even when ran `go mod tidy` anyways, I'll undo changes if possible

**Comment by alphaX86** [comment]
@leecalcote should we check this on k8s too? Or Docker alone is enough?

**Comment by alphaX86** [comment]
I noticed something... whenever a k8s cluster has improper endpoint, the GET request fails as below...

`Error: Get "https://127.0.0.1:55555/apis/apps/v1/namespaces/meshery/deployments": dial tcp 127.0.0.1:55555: connect: connection refused`
`Usage:`
`  mesheryctl system status [flags]`

May I know how and where it uses the GET request? That way I think I can create a debug message to make the user configure his k8s cluster properly
// @navendu-pottekkat

**Comment by leecalcote** [comment]
@umairnsr87, would you like to review this?

**Review by leecalcote — APPROVED** [review]

**Comment by umairnsr87** [comment]
@

> @umairnsr87, would you like to review this?

Seems good to me ... @alphaX86 @leecalcote

**Pull request merged**